### ISSUE TITLE
[type: new feature] adding integration test workflow

### ISSUE THREAD

**Issue by loongs-zhang** [brpc/brpc#1915]
**Is your feature request related to a problem? (你需要的功能是否与某个问题有关?)**
It is inevitable that there are problems in the code of manual review. Adding integration test workflow can effectively solve this problem.

**Describe the solution you'd like (描述你期望的解决方法)**
Adding integration test workflow, it is recommended to reuse the code below https://github.com/apache/incubator-brpc/tree/master/example . At present, I am not familiar with C++ and brpc. I hope that the master can complete it.

**Describe alternatives you've considered (描述你想到的折衷方案)**

**Additional context/screenshots (更多上下文/截图)**

**Comment by wwbmmm** [comment]
This is a good idea, brpc does not have integration tests yet. However, some existing unit tests are complex enough to cover integration scenarios, such as test/brpc_channel_unittest.cpp. Maybe we can extract these cases separately to make an integration test set.

**Comment by dragon-zhang** [comment]
Agree, the main purpose of integration test is to ensure that the new PR will not lead to the failure of important features.
